File: Advanced Programming Methods/Laboratoare/Lab/Lab10/src/sample/repository/NotaValidator.java
package sample.repository;

import sample.domain.Nota;

public class NotaValidator implements Validator<Nota> {
    public void validate(Nota nota){
        String err="";
        if(nota.getID().getKey()<1)
            err += "Id student invalid!\n";
        if(nota.getID().getValue()<1 || nota.getID().getValue()>14)
            err += "Numar tema invalid!\n";
        if(nota.getNota()<1 || nota.getNota()>10)
            err += "Nota invalida!\n";
        if(err!="")
            throw new ValidationException(err);
    }
}

File: Advanced Programming Methods/Laboratoare/Lab/Lab12/Lab12/Lab12/validator/NotaValidator.cs
using System;
using System.Collections.Generic;
using System.Text;
using Lab12.domain;

namespace Lab12.validator
{
    class NotaValidator:IValidator<Nota>
    {
        public NotaValidator() { }
        public void Validate(Nota nota)
        {
            String err = "";
            if (nota.Grade < 1 || nota.Grade > 10)
                err += "Nota invalida!\n";
            if (err != "")
                throw new ValidationException(err);
        }
    }
}
